### PR TITLE
Change how we add styles for Table Cell Selection

### PR DESCRIPTION
**Pull request by BryanValverdeU**
Changing how we are adding the styles when selecting table.

Using a style tag as recommended.

Most of the previously added Table cell selection tests are broken after this change, will fix them on future PR

+Added a property in the domEvent Plugin so we can save the table selection data.
+Modified the SelectionRangePath when in ShadowEdit from Path to Path[], that way if is table selection we just add the ranges from the added domEvent property.
+New Editor API so we can communicate with domEvent Plugin and save Table Selection (setTableSelection)
+Fixed tests from before Table Selection.

+Removed the styles used from Table Selection and now using new API.

asd

Splitted this PR in 3 for easier review

#748 
#749 
#750

### CODE REVIEW

File: packages/roosterjs-editor-types/lib/interface/IEditor.ts [hunk 1]
```diff
@@ -298,6 +299,11 @@ export default interface IEditor {
      */
     select(path: SelectionPath): boolean;
 
+    /**
+     * Select content using the Table Selection
+     */
+    setTableSelection(table: HTMLTableElement, coordinates: TableSelection): boolean;
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-types/lib/interface/IEditor.ts:305]: Prefer reuse existing select API, to let it accept TableSelection as parameter

File: packages/roosterjs-editor-types/lib/event/ShadowEditEvent.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export interface EnterShadowEditEvent extends BasePluginEvent<PluginEventType.En
     /**
      * The selection path of original editor content
      */
-    selectionPath: SelectionPath;
+    selectionPath: SelectionPath[];
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-types/lib/event/ShadowEditEvent.ts:17]: Do not change existing var definition, you can add additional property TableSelection

File: packages/roosterjs-editor-types/lib/corePluginState/LifecyclePluginState.ts [hunk 3]
```diff
@@ -45,5 +45,5 @@ export default interface LifecyclePluginState {
     /**
      * Cached selection path for original content
      */
-    shadowEditSelectionPath: SelectionPath;
+    shadowEditSelectionPath: SelectionPath[];
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-types/lib/corePluginState/LifecyclePluginState.ts:48]: Same here, do not change existing var def

File: packages/roosterjs-editor-dom/lib/selection/selectTable.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+/**
+ * Select a table
+ * @param table table to select
+ * @param coordinates first and last cell of the selection, if this parameter is null, instead of
+ * selecting, will unselect the table.
+ * @param document owner document of the table, if added will append a style element with styles to add to the table
+ * @returns TableSelectionRange if selected a new table, if unselecting the table, null
+ */
+export default function selectTable(
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/selection/selectTable.ts:14]: Suggest to put this into core package, and make it a core api, can either integrate together with select, or a separate one

File: packages/roosterjs-editor-dom/lib/selection/selectTable.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    table: HTMLTableElement,
+    coordinates?: TableSelection,
+    document?: Document
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/selection/selectTable.ts:17]: no need to pass in document

File: packages/roosterjs-editor-dom/lib/selection/selectTable.ts [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    return { firstCell: newFirst, lastCell: newLast };
+}
+
+function min(input1: number, input2: number) {
+    return input1 > input2 ? input2 : input1;
+}
+
+function max(input1: number, input2: number) {
+    return input1 < input2 ? input2 : input1;
+}
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/selection/selectTable.ts:138]: use Math.min, Math.max

**Comment by JiuqingSong** [comment]
Is there any existing code we can remove since the selection functionality is moved into editor core?

**Comment by BryanValverdeU** [comment]
> Is there any existing code we can remove since the selection functionality is moved into editor core?

Probably most of the utils inside of the table selection will be removed if we can do this change.

**Pull request closed without merging**